src/components/ListItem.jsx

```diff
@@ -1,5 +1,72 @@
+import Checkbox from '@mui/material/Checkbox';
+import { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item }) {
+	// Get the initial checked state from localStorage or default to false
+	const [checked, setChecked] = useState(() => {
+		const storedChecked = localStorage.getItem(`checked-${item.name}`);
+		return storedChecked ? JSON.parse(storedChecked) : false;
+	});
+
+	// Function to check if 24 hours have passed
+	const has24HoursPassed = (dateLastPurchased) => {
+		const purchaseTime = dateLastPurchased.getTime(); // Time in milliseconds
+		const currentTime = new Date().getTime(); // Current time in milliseconds
+		const oneDayInMs = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
+		return currentTime - purchaseTime >= oneDayInMs;
+	};
+
+	// Run the effect when the component mounts
+	useEffect(() => {
+		if (item.dateLastPurchased) {
+			const purchaseDate = item.dateLastPurchased.toDate(); // Convert Firebase timestamp to Date
+			if (checked) {
+				// console.log('Checking if 24 hours passed since last purchase');
+				const is24HoursPassed = has24HoursPassed(purchaseDate);
+				// console.log(`Has 24 hours passed: ${is24HoursPassed}`);
+
+				// If 24 hours have passed, uncheck the checkbox associated with the item.
+				if (is24HoursPassed) {
+					// console.log('Unchecking item because 24 hours have passed');
+					setChecked(false);
+					localStorage.setItem(`checked-${item.name}`, JSON.stringify(false));
```

Comment:
**nitpick ( non-blocking ):** Maybe passing the checked state into since its updating anyway.